### PR TITLE
[SW-1740] Add parallelism option to GridSearch

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala [hunk 1]
```diff
@@ -391,7 +391,12 @@ trait H2OGridSearchParams extends H2OCommonSupervisedParams {
   private val stoppingMetric = new Param[String](this, "stoppingMetric", "Stopping Metric")
   private val selectBestModelBy = new Param[String](this, "selectBestModelBy", "Select best model by specific metric." +
     "If this value is not specified that the first model os taken.")
-
+  private val parallelism = new IntParam(this,
+    "parallelism",
+    """Level of model-building parallelism, the possible values are:
+      | 0 -> H2O selects parallelism level based on cluster configuration, such as number of cores
+      | 1 -> Sequential model building, no parallelism
+      | n>1 -> n models will be build in parallel if possible""".stripMargin)
```
Comment (mn-mikke) [inline ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala]: `be build` -> `be built`?

File: py/src/ai/h2o/sparkling/ml/params/H2OGridSearchParams.py [hunk 2]
```diff
@@ -78,6 +78,15 @@ class H2OGridSearchParams(H2OCommonSupervisedParams):
         "Specifies the metric which is used for comparing and sorting the models returned by the grid.",
         H2OTypeConverters.toEnumString("ai.h2o.sparkling.ml.algos.H2OGridSearchMetric"))
 
+    parallelism = Param(
+        Params._dummy(),
+        "parallelism",
+        """Level of model-building parallelism, the possible values are:
+           0 -> H2O selects parallelism level based on cluster configuration, such as number of cores
+           1 -> Sequential model building, no parallelism
+           n>1 -> n models will be build in parallel if possible""",
```
Comment (mn-mikke) [inline py/src/ai/h2o/sparkling/ml/params/H2OGridSearchParams.py]: `be build` -> `be built`?

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala (hunk 1 above)
Comment (jakubhava) [reply ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala]: thanks, fixed

File: py/src/ai/h2o/sparkling/ml/params/H2OGridSearchParams.py (hunk 2 above)
Comment (jakubhava) [reply py/src/ai/h2o/sparkling/ml/params/H2OGridSearchParams.py]: thanks!

**Pull request merged**